src/higlass_schema/utils.py

```diff
@@ -1,38 +1,67 @@
 from typing import Any, Dict, TypeVar
 
+import pydantic_core.core_schema as core_schema
 from pydantic import BaseModel, TypeAdapter
-
-
-def simplify_schema(root_schema: Dict[str, Any]) -> Dict[str, Any]:
-    """Lift defintion reference to root if only definition"""
-    # type of root is not a reference to a definition
-    if "$ref" not in root_schema:
-        return root_schema
-
-    defs = list(root_schema["$defs"].values())
-    if len(defs) != 1:
-        return root_schema
-
-    return defs[0]
+from pydantic._internal._core_utils import CoreSchemaOrField, is_core_schema
```

Comment:
Unfortunate to have to rely on private stuff. In case you want to vendor it in:

```python
from typing import Union
from typing_extensions import TypeGuard

from pydantic_core import core_schema, CoreSchemaField

CoreSchemaField = Union[
    core_schema.ModelField, core_schema.DataclassField, core_schema.TypedDictField, core_schema.ComputedField
]

CoreSchemaOrField = Union[core_schema.CoreSchema, CoreSchemaField]

_CORE_SCHEMA_FIELD_TYPES = {'typed-dict-field', 'dataclass-field', 'model-field', 'computed-field'}

def is_core_schema(
    schema: CoreSchemaOrField,
) -> TypeGuard[core_schema.CoreSchema]:
    return schema['type'] not in _CORE_SCHEMA_FIELD_TYPES
```